lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -130,20 +130,19 @@ def ortho_transformation(zfront, zback):
 
 def proj_transform_vec(vec, M):
     vecw = np.dot(M, vec)
-    w = vecw[3]
-    # clip here..
-    txs, tys, tzs = vecw[0]/w, vecw[1]/w, vecw[2]/w
-    return txs, tys, tzs
+    vecw /= vecw[3]
+    return vecw[0:3]
 
 def proj_transform_vec_clip(vec, M):
     vecw = np.dot(M, vec)
-    w = vecw[3]
-    # clip here.
-    txs, tys, tzs = vecw[0] / w, vecw[1] / w, vecw[2] / w
+    # Determine clipping before rescaling
     tis = (0 <= vecw[0]) & (vecw[0] <= 1) & (0 <= vecw[1]) & (vecw[1] <= 1)
     if np.any(tis):
         tis = vecw[1] < 1
-    return txs, tys, tzs, tis
+    vecw /= vecw[3]
+    # Integrating tis in the numpy array for optimization purposes
+    vecw[3, :] = tis
+    return vecw
```

Comment:
https://github.com/matplotlib/matplotlib/pull/6085#pullrequestreview-51068714 by @eric-wieser 

> Storing coordinates in the first axis of the array is kind of unusual (see how np.linalg and scipy.misc.imread behave).
> 
> Perhaps these should all be xyz[..., 0] instead of xyz[0]?

Well in the current case `xyz[0]` for default C aligned arrays will produce a contiguous array while `xyz[..., 0]` won't unless I'm mistaken, which might have a performance cost (I haven't tested it).  

Is it essentially the difference between `np.meshgrid` with `indexing="xy"` and `indexing="ij"`? 

I have not changed this, if you think this is critical I will.

Reply:
You're correct about contiguity, but the choice of internal storage order and external indexing order is orthogonal.

I'll have to come back to this later to decide if it's important